grow/preprocessors/gulp.py

```diff
@@ -58,7 +59,17 @@ def _kill_child_process():
     for process in _child_processes:
         try:
             process.terminate()
-            process.wait()
         except OSError:
             # Ignore the error.  The OSError doesn't seem to be documented(?)
             pass
+
+    # Give the process some time to finish by itself.
+    time.sleep(3)
```

Comment:
it looks like you're always enforcing a 3-second lag on every shutdown, even if the child process takes less time than that to shut down. i'd suggest sleeping on some smaller interval and checking if the process is still running, and force killing it after the max of 3 seconds.